### DISCUSSION TITLE
how to submit in c++ 20

### DISCUSSION THREAD

**Discussion by Harshwardhan4k** [cpeditor/cpeditor#1055] [Q&A] [answered]
i have latest MinGw for c++ 20 , but still not able to submit C++ 20 on codeforces, via CF tools, it is by default submitting in c++17

**Comment by ouuan** [comment, marked as answer]
Until we support other submitting tools or CF Tool supports submitting in C++20, you need to manually submit. The language list is hardcoded at https://github.com/xalanq/cf-tool/blob/4aff8682ad0d1e14155c9744a3f27780576bb11e/client/langs.go, but I'm not sure how we can modify it and whether it's enough to modify this file to get c++20 submission working.

**Reply by useredsa** [reply]
Actually, you can write the code for c++20 manually in `.cf/config`. It does not appear in the list but it will use whichever code is set in the config file. The list purpose is helping you generating this config file.

**Comment by ouuan** [comment]
BTW, CP Editor doesn't choose the language. It's set in the CF Tool config.

**Comment by markjouh** [comment]
Hello, I had this issue and resolved it by installing this [fork](https://github.com/izlyforever/cf-tool) of CF Tool, which just includes some changes to make the tool more up-to-date. It supports C++20.
